flink-ml-core/src/main/java/org/apache/flink/ml/util/QuantileSummary.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.util;
+
+import org.apache.flink.util.Preconditions;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.apache.flink.util.Preconditions.checkState;
+
+/**
+ * Helper class to compute an approximate quantile summary. This implementation is based on the
+ * algorithm proposed in the paper: "Space-efficient Online Computation of Quantile Summaries" by
+ * Greenwald, Michael and Khanna, Sanjeev. (https://doi.org/10.1145/375663.375670)
+ */
+public class QuantileSummary {
+
+    private static final int BUFFER_SIZE = 10000;
+    private static final int COMPRESS_THRESHOLD = 10000;
+
+    private double relativeError;
+    private long count;
+    private List<StatsTuple> sampled;
+    private List<Double> headBuffer = new ArrayList<>(BUFFER_SIZE);
+
+    /**
+     * * QuantileSummaries Constructor.
+     *
+     * @param relativeError The target relative error.
+     */
+    public QuantileSummary(double relativeError) {
```

Comment:
As from LSH and RobustScaler, the constructor needs to take both `relativeError` and `compressThreshold` as input parameters.

Reply:
I will add another public constructor with both `relativeError` and `compressThreshold` but also keep the construct with only `relativeError` meanwhile, because in most cases users do not need to worry about the `compressThreshold`, so do LSH and RobustScaler.